### PR TITLE
Code optimization; new functions

### PR DESCRIPTION
**Pull request by kaushalmodi**
- Use of `funcall` to call the right command based on if the user uses
  `pdf-tools` package or not
- Refine `interleave--doc-view-kill-proc-and-buffer` fn so that the
  `doc-view-kill-proc` is called only if the major mode is
  `doc-view-mode`.
- Added `*interleave--pdf-buffer*`; renamed _interleave--org-buf_
  to _interleave--org-buffer_
- Use of `switch-to-buffer` and `switch-to-buffer-other-window` instead
  of `(other-window 1)` for more accuracy (in the event the user has
  opened a buffer other than the pdf or org notes file in the "other"
  window).
- Added commands to synchronize the pdf page with the page displayed in
  current org notes buffer, to sync the pdf page to the next/previous
  notes page:
  - interleave--sync-pdf-page-current
  - interleave--sync-pdf-page-previous
  - interleave--sync-pdf-page-next
- Creates variable handles `interleave-map` and
  `interleave-pdf-mode-map` so a user can customize the key bindings in
  them without having to modify the source.
- Changed the lighter strings for both pdf and org buffer minor modes to " ≡"

### CODE REVIEW

File: interleave.el [hunk 1]
```diff
@@ -38,28 +39,48 @@
...
+(defvar *interleave--pdf-buffer* nil "PDF buffer associated with the notes buffer")
 
 (defvar interleave--window-configuration nil
   "Variable to store the window configuration before interleave mode was
 enabled.")
 
+(if (featurep 'pdf-view) ; if `pdf-tools' is installed
+    (progn
+      ;; Function wrapper for the macro `pdf-view-current-page'
+      (defconst interleave--pdf-current-page-fn                 (lambda () (pdf-view-current-page)))
```
Comment (kaushalmodi) [inline interleave.el:71]: @rudolfochrist  This was the reason why your earlier approach to use the change page hook in pdf-view failed. But I realized that at the very end after I changed everything in this commit. 

If you had wrapped `pdf-view-current-page` in a `lambda` wrapper, your approach would have worked too. But I did not know that then.

**Comment by kaushalmodi** [comment]
I need to understand why I see this note on the pull request page: "Can't automatically merge. Don't worry, you can still create the pull request."

File: interleave.el [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 ;; Author: Sebastian Christ <rudolfo.christ@gmail.com>
 ;; URL: https://github.com/rudolfochrist/interleave
-;; Version: 0.2.0
```
Comment (kaushalmodi) [inline interleave.el:5]: The pull request is comparing against a slightly older version. So I am missing some `git` step. Let me know if this is obvious to you. What do I need to do once you have merged the pull request and both of our "master"s are in sync? Do I need to hard reset my fork to your master (How?)?

Comment (rudolfochrist) [inline interleave.el:5]: I think rebasing to upstream/master will do the trick.

File: interleave.el [hunk 3]
```diff
@@ -118,139 +181,146 @@ to PAGE. It narrows the subtree when found."
...
+(defun interleave--sync-pdf-page-current ()
+  "Synchronize the page in the pdf buffer to be the same as the page in
+the current narrowed down notes view."
+  (interactive)
+  (interleave--switch-to-org-buffer)
+  (let ((pdf-page))
+    (save-excursion
+      (org-narrow-to-subtree)
+      (goto-char (point-min))
+      (search-forward-regexp "Notes for page \\(.*\\)")
```
Comment (rudolfochrist) [inline interleave.el]: Can you search for the `interleave_page_note` property instead. The headline should be changeable. Thanks.

**Comment by kaushalmodi** [comment]
Changed the search to org heading properties instead of `"Notes for page"` string in https://github.com/kaushalmodi/interleave/commit/3fff9d9001989a5ae9bf2666bf6c32ceef930ff3

Improved robustness in https://github.com/kaushalmodi/interleave/commit/e5a8304c8a33d79f56d6ac7afbbbbafb57db828b

**Comment by rudolfochrist** [comment]
Works like a charm. Especially the buffer synchronization is a nice enhancement. Thank you.

**Pull request merged**